clash-prelude/tests/Clash/Tests/Resize.hs

```diff
@@ -26,16 +45,146 @@ indexProp Proxy v =
 indexFailProp ::
   forall a b.
   ((b <= a), KnownNat a, KnownNat b) =>
-  Proxy b -> Index a -> Property
+  Proxy b -> Index a -> Q.Property
 indexFailProp Proxy v =
   let checked = Resize.checkedResize @a @b v in
   if toInteger v > toInteger (maxBound @(Index b)) then
     expectExceptionNoX checked
   else
     discard
 
+maybeResizePropT ::
+  forall f a b .
+  ( Integral (f a)
+  , Bounded (f a)
+  , Show (f a)
+  , Integral (f b)
+  , Bounded (f b)
+  , Show (f b)
+  , Resize.Resize f
+  , KnownNat a
+  , KnownNat b
+  ) =>
+  Proxy (f a) ->
+  Proxy (f b) ->
+  H.PropertyT IO ()
+maybeResizePropT _ _ = do
+  let
+    minFa = fromIntegral (minBound @(f a)) :: Integer
+    maxFa = fromIntegral (maxBound @(f a)) :: Integer
+    minFb = fromIntegral (minBound @(f b)) :: Integer
+    maxFb = fromIntegral (maxBound @(f b)) :: Integer
+
+  H.footnote $ "minFa: " <> show minFa
+  H.footnote $ "maxFa: " <> show maxFa
+  H.footnote $ "minFb: " <> show minFb
+  H.footnote $ "maxFb: " <> show maxFb
+
+  input <- H.forAll $ Gen.integral (Range.linear minFa maxFa)
```

Comment:
Wouldn't these be rather biased to zero? I.e., not producing interesting cases?
```suggestion
  input <- H.forAll $ Gen.integral (Range.constant minFa maxFa)
```

Reply:
Yes, it would. I wasn't aware of `constant`! Actually, they both suck. They basically have zero chance of producing interesting edge cases.

Reply:
We know what the edge cases are, so we could produce primarily edge cases. [`Gen.frequency`](https://hackage.haskell.org/package/hedgehog-1.5/docs/Hedgehog-Gen.html#v:frequency).

You know, now that I look at it, the doc for `frequency` only says it shrinks towards the first generator. I really thought it also only considers earlier items on the list for small sizes. If I now look at the implementation, it would appear I am wrong about that.

Reply:
Would you mind if I skip this (=> `frequency`. I've changed to `constant`.)? I don't want to spend much more effort. Or rather, I think the current situation is "good enough".

Reply:
Go ahead. It's good enough for me with `constant`, I just continued on from your assertion that it still sucked :-D.